### PR TITLE
scheduler/extender: collapse client wiring onto standard approach

### PR DESCRIPTION
**Pull request by enj**
Signed-off-by: Monis Khan <mok@microsoft.com>

/kind cleanup

```release-note
NONE
```

This is mostly to avoid another #114588 in the future.
xref: #115315

I think we could reasonably mark `TLSConfigFor` as deprecated.

The only other use is:

https://github.com/kubernetes/kubernetes/blob/161b00c639c110e66f8a97cdb95eea3dbe0a25e0/pkg/kubelet/certificate/transport.go#L82-L83

Which handles rotation correctly:

https://github.com/kubernetes/kubernetes/blob/161b00c639c110e66f8a97cdb95eea3dbe0a25e0/pkg/kubelet/certificate/transport.go#L147-L152

xref: https://github.com/kubernetes-incubator/bootkube/pull/663#issuecomment-318506493

/assign @liggitt @aojea

### CODE REVIEW

File: pkg/scheduler/extender.go [hunk 1]
```diff
@@ -70,35 +69,23 @@ func makeTransport(config *schedulerapi.Extender) (http.RoundTripper, error) {
...
-		return nil, err
-	}
-	if tlsConfig != nil {
-		return utilnet.SetTransportDefaults(&http.Transport{
-			TLSClientConfig: tlsConfig,
-		}), nil
-	}
-	return utilnet.SetTransportDefaults(&http.Transport{}), nil
+	cfg.Timeout = config.HTTPTimeout.Duration
+	return restclient.HTTPClientFor(&cfg)
```
Comment (aojea) [inline pkg/scheduler/extender.go:73]: this adds the transports Wrappers, and previously it just created a transport from the TLSConfig, no?

Comment (enj) [reply pkg/scheduler/extender.go:73]: Only the `DebugWrappers` should be used, which sounds like a good thing?

Comment (aojea) [reply pkg/scheduler/extender.go:73]: ah, right, we pass an empty rest.Config , that makes sense

/lgtm

**Comment by liggitt** [comment]
lgtm, would like an ack from a scheduler reviewer... not sure how good our test coverage on this is

/assign @alculquicondor

**Comment by alculquicondor** [comment]
There is an integration test, but something like the timeout configuration could be broken by this PR and we wouldn't know.

@enj any chance you can improve coverage? Maybe something can be covered in unit tests?

https://github.com/kubernetes/kubernetes/blob/master/test/integration/scheduler/extender/extender_test.go

**Comment by moficodes** [comment]
@alculquicondor @liggitt 

1.28 Bug Triage Shadow here.

Just checking-in, if this is still on track for the 1.28 release?

**Comment by alculquicondor** [comment]
This is in @enj's court to address at the moment.

**Comment by furkatgofurov7** [comment]
@enj hey again! Bug Triage Team ping

Can you please rebase the PR to move this forward? 

Please be aware the code freeze is starting [01:00 UTC Wednesday 19th July 2023 / 18:00 PDT Tuesday 18th July 2023](https://everytimezone.com/s/72ee8496) (about less than 2 weeks from now).

**Comment by moficodes** [comment]
Week 9 update: Code Freeze is approaching!

Hey! We are currently in Week 9, and Code Freeze is just over a week away. It will begin at 01:00 UTC on Wednesday, 19th July 2023 / 18:00 PDT on Tuesday, 18th July 2023. Please ensure all necessary submissions are made before the deadline. Let me know if you need any assistance. :)

**Comment by moficodes** [comment]
Hello @alculquicondor  ! 👋🏼
Bug triage for the 1.29 release cycle is here!
This issue hasn't been updated for a long time, so I wanted to check what the status is. The code freeze will start ([01:00 UTC Wednesday 1st November 2023 / 18:00 PDT Tuesday 31st October 2023](https://everytimezone.com/s/3b849751)), which is about 2 weeks from now. And while there is still plenty of time, we want to make sure that every PR has a chance to be merged on time.

As the issue is tagged for 1.29, is it still planned for that release?

**Comment by alculquicondor** [comment]
I'm still waiting for @enj to address the questions about testing.

**Comment by moficodes** [comment]
Hello @enj ! 👋🏼
Bug triage for the 1.29 release cycle is here!
This issue hasn't been updated for a long time, so I wanted to check what the status is. The code freeze will start ([01:00 UTC Wednesday 1st November 2023 / 18:00 PDT Tuesday 31st October 2023](https://everytimezone.com/s/3b849751)), which is this week. We want to make sure that every PR has a chance to be merged on time.

As the issue is tagged for 1.29, is it still planned for that release?

**Comment by dims** [comment]
/milestone clear

thread on slack: https://kubernetes.slack.com/archives/C09TP78DV/p1699885081516409

**Pull request closed without merging**